document/JS.bs

```diff
@@ -0,0 +1,921 @@
+</div>
+
+<div algorithm>
+    The <dfn for="Table">get(index)</dfn> method, when invoked, performs the following steps:
+    1. Let |values| be the table instance's \[[Values]] internal slot.
+    1. Let |size| be the length of |values|.
+    1. If |index| &ge; |size|, throw a [=RangeError=] exception.
+    1. Return |values|[|index|].
+</div>
+
+<div algorithm>
+    The <dfn for="Table">set(index, value)</dfn> method, when invoked, performs the following steps:
+    1. Let |tableaddr| be the table instance's \[[Table]] internal slot.
+    1. Let |values| be the table instance's \[[Values]] internal slot.
+    1. If |value| is null, let |elem| be an empty [=function element=].
+    1. Otherwise, let |funcaddr| be |value|.\[[FunctionAddress]].
+    1. Let |store| be the current agent cluster's [=associated store=].
+    1. Let |store| be [=write_table=](|store|, |tableaddr|, |index|, |funcaddr|).
+    1. If |store| is [=error=], throw a [=RangeError=] exception.
+    1. Set the current agent cluster's [=associated store=] to |store|.
+    1. Set |values|[|index|] to |value|.
+    1. Return undefined.
+</div>
+
+<h3 id="exported-function-exotic-objects">Exported Functions</h3>
+
+<pre class='idl'>
+[NoInterfaceObject]
+interface ExportedFunction : Function { };
+</pre>
+
+A WebAssembly function is made available in JavaScript as a <dfn>Exported Function
+Exotic Object</dfn>, i.e., an {{ExportedFunction}} instance. ExportedFunctions are
+[=Built-in Function Objects=] with a \[[FunctionAddress]] internal slot and special
+behavior for \[[Call]] and \[[Construct]]. This slot holds a [=function address=]
+relative to the current agent cluster's [=associated store=].
+
+<div algorithm>
+  To <dfn>create an Exported Function</dfn> from a WebAssembly [=function address=] |funcaddr|, perform the following steps:
+
+    1. Let |steps| be "[=call an Exported Function|call the Exported Function=] |funcaddr| with arguments."
+    1. Let |realm| be the [=current Realm=].
+    1. Let |function| be [=CreateBuiltinFunction=](|realm|, |steps|, [=%FunctionPrototype%=], &laquo; \[[FunctionAddress]]).
+    1. Set |function|.\[[FunctionAddress]] to |funcaddr|.
+    1. Let |store| be the current agent cluster's [=associated store=].
+    1. Let |functype| be [=type_func=](|store|, |funcaddr|).
+    1. Let [|arguments|] → [|results|] be |functype|.
+    1. Let |arity| be the length of |arguments|.
+    1. Perform ! [=DefinePropertyOrThrow=](|function|, `"length"`, PropertyDescriptor {\[[Value]]: |arity|, \[[Writable]]: false, \[[Enumerable]]: false, \[[Configurable]]: true}).
+    1. Let |index| be the [=function index=] associated with |funcaddr|.  <!-- TODO: Define more cleanly? -->
+    1. Perform ! [=SetFunctionName=](|function|, ! [=ToString=](|index|)).
+    1. Return |function|.
+</div>
+
+<div algorithm>
+  To <dfn>call an Exported Function</dfn> |funcaddr| with arguments |argValues|, perform the following steps:
```

Comment:
Although usually I don't declare the types of my arguments in spec algorithms, I think here it might help. I'm guessing this is a WebAssembly function Address and a list of ... JS values?